### ISSUE TITLE
Proposal: docker build + docker create without image affinities

### ISSUE THREAD

**Issue by aluzzardi** [docker/swarm#743]
Right now, in order to build in the cluster and run a container one must:

```
docker build -t foo .
docker create -e affinity:image==foo foo
```

That's because `foo` ends up on one machine but the container using `foo` might end up on a different one since swarm has no idea that the image is local and cannot be pulled from the registry

How about this:

1/ Try a create. It fails with 404
2/ Try a pull. It fails with something like `image not found`
3/ Check if the image exists in the cluster.
4/ Retry with a soft-affinity (but **don't** store it in the `ContainerConfig`)
5/ If it still fails, then there is a problem with the image, abort
6/ If it works, celebrate

**Comment by aluzzardi** [comment]
@vieux Thoughts?

**Comment by vieux** [comment]
Great idea.

**Comment by electrical** [comment]
Sounds a very good solution for the problem i'm having now :-)

**Comment by marc-** [comment]
How about locate the image and replicate it across cluster (or copy to specific node on demand)?

**Comment by coffenbacher** [comment]
This would be awesome for my workflow. Good idea, can't wait!

**Comment by abronan** [comment]
closed by #971
